### PR TITLE
check for expired or removed k-lines on startup

### PR DESCRIPTION
**Pull request by examknow**

### CODE REVIEW

File: beryllia/__init__.py [hunk 1]
```diff
@@ -109,6 +111,43 @@ async def _is_oper(self, nickname: str):
         })
         return whois_line.command == RPL_WHOISOPERATOR
 
+    async def _find_unset(self):
+        # check stats k (locally set kline) and g (kline set on another server)
+        # to see if anything was unset while we were gone
+        klines        = await self.database.list_klines()
```
Comment (jesopo) [inline beryllia/__init__.py]: can this be `known_klines` to contrast with `active_klines`

File: beryllia/__init__.py [hunk 2]
```diff
@@ -109,6 +111,43 @@ async def _is_oper(self, nickname: str):
...
+            stats_line = await self.wait_for({
+                stats_kline, stats_end
+            })
+            if stats_line.command == RPL_STATSKLINE:
+                active_klines.append(f"{stats_line.params[4]}@{stats_line.params[2]}")
+            else:
+                break
+
+        for id, mask in klines:
+            if not mask in active_klines:
```
Comment (jesopo) [inline beryllia/__init__.py]: it'd be more performant to have `active_klines` be a set, and convert the result of `self.database.list_klines()` in to a dictionary of `{mask: id}`, and then find unset kline masks by doing `unset_klines = set(known_klines.keys()) - active_klines`

Comment (examknow) [reply beryllia/__init__.py]: but with this way we wouldn't have the id of the k-line that we need to remove it from the database; we'd only have the masks.

Comment (jesopo) [reply beryllia/__init__.py]: that's why you have a dict to look the mask up in

Comment (examknow) [reply beryllia/__init__.py]: right...

File: beryllia/__init__.py [hunk 3]
```diff
@@ -109,6 +111,43 @@ async def _is_oper(self, nickname: str):
...
+        # check stats k (locally set kline) and g (kline set on another server)
+        # to see if anything was unset while we were gone
+        known_klines  = await self.database.list_klines()
+        active_klines = set()
+        await self.send(build("STATS", ["k"]))
+
+        stats_kline = Response(RPL_STATSKLINE, [SELF, ANY])
+        stats_end   = Response(RPL_ENDOFSTATS, [SELF, ANY])
+        #:lithium.libera.chat 216 sandcat sandcat :k 1.2.3.4 * * :piss off m8 (2021/8/23 20.24)|(launchd!launchd@localhost{launchd})
+        #:lithium.libera.chat 219 sandcat sandcat :k :End of /STATS report
```
Comment (jesopo) [inline beryllia/__init__.py]: these examples appear to be incorrect. from solanum code it looks like it should be 
```python
#:lithium.libera.chat 216 sandcat k 1.2.3.4 * * :piss off m8 (2021/8/23 20.24)|(launchd!launchd@localhost{launchd})
#:lithium.libera.chat 219 sandcat k :End of /STATS report

**Pull request closed without merging**